pkg/scheduler/api/devices/nvidia/vgpu/mig.go

```diff
@@ -0,0 +1,328 @@
+/*
+Copyright 2025 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgpu
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"volcano.sh/volcano/pkg/scheduler/api/devices/config"
+)
+
+type MIGFactory struct{}
+
+func (f MIGFactory) TryAddPod(gd *GPUDevice, mem uint, core uint) (bool, string) {
+	found, dev, usedMem := findMatch(gd.UUID, mem, gd.MigUsage, gd.MigTemplate)
+	if !found {
+		return false, ""
+	}
+
+	gd.UsedNum++
+	gd.UsedMem += usedMem
+	gd.UsedCore += core
+	return true, dev
+}
+
+func (f MIGFactory) AddPod(gd *GPUDevice, mem uint, core uint, podUID string, devID string) error {
+	group, index, err := decodeMIGID(devID)
+	if err != nil {
+		klog.ErrorS(err, "Failed to add pod")
+		return err
+	}
+
+	_, ok := gd.PodMap[podUID]
+	if !ok {
+		gd.PodMap[podUID] = &GPUUsage{
+			UsedMem:  0,
+			UsedCore: 0,
+		}
+	} else {
+		return nil
+	}
+	usedMem := addMigUsed(gd, group, index)
+	gd.UsedNum++
+	gd.UsedMem += usedMem
+	gd.UsedCore += core
+
+	gd.PodMap[podUID].UsedMem += usedMem
+	gd.PodMap[podUID].UsedCore += core
+
+	klog.V(4).Infoln("add Pod: ", podUID, usedMem, gd.PodMap[podUID].UsedMem, gd.PodMap[podUID].UsedCore)
+	return nil
+}
+
+func (f MIGFactory) SubPod(gd *GPUDevice, mem uint, core uint, podUID string, devID string) {
+	groupName, index, err := decodeMIGID(devID)
+	if err != nil {
+		klog.ErrorS(err, "Failed to sub pod")
+		return
```

Comment:
```suggestion
		return err
```

Reply:
the caller function SubResource does not return value.